FinSearchUnified/Tests/Helper/StaticHelperTest.php

```diff
@@ -430,5 +468,146 @@ private function updateParentCategoryName(Category $parent, Category $categoryMo
         if ($parent->getParent() !== null) {
             $this->updateParentCategoryName($parent->getParent(), $parent, $restore);
         }
+
+        Shopware()->Models()->flush();
+    }
+
+    public function finSmartDidYouMeanProvider()
+    {
+        return [
+            'didYouMeanQuery and originalQuery are not present' => [null, null, null, null, null, null],
+            'Attribute type of queryString is forced' => [null, 'originalQuery', 'forced', null, null, null],
+            'didYouMeanQuery is present but has no value' => ['', null, 'improved', null, null, null],
+            'originalQuery is present but has no value' => [null, '', 'improved', null, null, null],
+            'didYouMeanQuery is present' => [
+                'didYouMeanQuery',
+                'originalQuery',
+                'improved',
+                'did-you-mean',
+                'didYouMeanQuery',
+                ''
+            ],
+            'queryString type is improved' => [
+                null,
+                'originalQuery',
+                'improved',
+                'improved',
+                'queryString',
+                'originalQuery'
+            ],
+            'queryString type is corrected' => [
+                null,
+                'originalQuery',
+                'corrected',
+                'corrected',
+                'queryString',
+                'originalQuery'
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider finSmartDidYouMeanProvider
+     *
+     * @param string $didYouMeanQuery
+     * @param string $originalQuery
+     * @param string $queryStringType
+     * @param string $expectedType
+     * @param string $expectedAlternativeQuery
+     * @param string $expectedOriginalQuery
+     */
+    public function testFinSmartDidYouMean(
```

Comment:
Rename this to `testParsesSmartDidYouMeanData`. Otherwise it is hard to tell what this is actually testing.